### DISCUSSION TITLE
Cannot find module

### DISCUSSION THREAD

**Discussion by Sokwva** [wxt-dev/wxt#950] [General]
```shell
# pnpm run dev

WXT 0.19.7                                                                                                                                                                                                19:49:15
× Command failed after 2.347 s                                                                                                                                                                            19:49:17

 ERROR  Cannot find module '@/Declare/OptionConf'                                                                                                                                                         19:49:17
Require stack:
- E:\main\src\Core\CoreLibs\OptionDriver.ts

  Require stack:
  - src\Core\CoreLibs\OptionDriver.ts
  at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
  at Function.resolve (node:internal/modules/helpers:187:19)
  at _resolve (node_modules\.pnpm\jiti@1.21.6\node_modules\jiti\dist\jiti.js:1:241814)
  at jiti (node_modules\.pnpm\jiti@1.21.6\node_modules\jiti\dist\jiti.js:1:244531)
  at E:/main/src/Core/CoreLibs/OptionDriver.ts:1:184
  at evalModule (node_modules\.pnpm\jiti@1.21.6\node_modules\jiti\dist\jiti.js:1:247313)
  at jiti (node_modules\.pnpm\jiti@1.21.6\node_modules\jiti\dist\jiti.js:1:245241)
  at E:/main/src/Core/CoreUtils.ts:3:21
  at evalModule (node_modules\.pnpm\jiti@1.21.6\node_modules\jiti\dist\jiti.js:1:247313)
  at jiti (node_modules\.pnpm\jiti@1.21.6\node_modules\jiti\dist\jiti.js:1:245241)

 ELIFECYCLE  Command failed with exit code 1.
```

after upgrade wxt to 0.19,it throws error like this.

**Comment by aklinker1** [comment]
Don't run code outside the `main` function of your entrypoint. Without seeing your entrypoint, can tell you what's wrong. That error message is horrible though, I can improve it.

**Reply by Sokwva** [reply]
my repository is https://github.com/Sokwva/acfun-helper ,and my entrypoints located in src/Core/Entry ,i doubt if my entrypoint path too deep.

**Reply by aklinker1** [reply]
WXT introduced a new `modules` directory, which it tries to import files from. These files are meant to modify the build process. Since you have a folder at `src/Modules`, WXT thinks the files in it are meant to be WXT modules. To fix this, tell WXT that the modules directory should be another directory:

https://github.com/Sokwva/acfun-helper/blob/734d90e836702a1681aca785a99d71e483ccfbaa/wxt.config.ts#L45

```diff
  entrypointsDir: "./Core/Entry",
+ modulesDir: "./WxtModules",
  publicDir: "./Asset",
```

**Reply by aklinker1** [reply]
Interesting that WXT found `src/Modules`, it should be case-sensitive... it should look for `src/modules`. Will open a bug for that.

**Reply by Sokwva** [reply]
oh,that's interesting (ゝ∀･).

**Reply by aklinker1** [reply]
I've retroactively added docs about this to the upgrade page: https://wxt.dev/guide/upgrade-guide/wxt.html#v0-18-0-rarr-v0-18-5

Hopefully that will notify others of this addition.
